Add kubectl port-froward to integration tests

mabdelgowa opened the pull request:
Add kubectl port-froward to integration tests

aabouzaid reviewed (changes requested):
Thanks for your contribution 🙌
I've left a couple of comments.

aabouzaid commented:
Also, don't forget to rebase to get the latest updates.

mabdelgowa commented:
Hi eng @aabouzaid 👋👋👋👋👋👋
I rebased my forked repo and updated the testsuite-core.yaml file with two ranges (one for port forward cmmand and another one for verification ) . recheck it right now

aabouzaid commented on charts/camunda-platform-alpha/test/integration/testsuites/vars/files/testsuite-core.yaml:
```diff
@@ -305,3 +311,67 @@ testcases:
+    type: exec
+    range:
+    - component: zeebe-gateway
+      script: "{{.portforward}}/camunda-zeebe-gateway 26500:26500" 
```
@mabdelgowa, this will not work as port forwarding is a blocking command, and it will not return (will wait forever as it works).

So you can port forward and check at the same time in the `script` section something like this:

```
[...]
 range:
  - component: "Zeebe Gateway"
    serviceName: "camunda-zeebe-gateway"
    servicePort:  "26500"
    [...]
script: |
  (kubectl port-forward svc/{{ .serviceName }} {{ .servicePort }}:{{ .servicePort }} &
  curl localhost:{{ .servicePort }})
```

Please ensure to test it locally first to ensure the proposed change works as expected (you can test it with any service, not necessarily Camunda deployment).

mabdelgowa commented:
@aabouzaid I tested it locally after editing the script line to be like this (kubectl port-forward svc/{{.value.serviceName}} {{.value.servicePort}}:{{.value.servicePort}} > /dev/null 2>&1 &  curl localhost:{{.value.servicePort}}) and it passed for the opened services as it in the attached screenshots (the services which I used was camunda-zeebe-gateway --> 8080 not 26500  & camunda-operate --> 8081 ) it was only for testing 
![image](https://github.com/user-attachments/assets/13df2cc5-143a-49e0-b068-efd925282eae)
![image](https://github.com/user-attachments/assets/27cf83e2-62b0-4343-a3b4-9b5a7addad1d)

aabouzaid commented on charts/camunda-platform-alpha/test/integration/testsuites/vars/files/testsuite-core.yaml:
```diff
@@ -305,3 +311,62 @@ testcases:
+    - component: zeebe-gateway
+      serviceName: camunda-zeebe-gateway
+      servicePort: 26500
+      script: "kubectl port-forward svc/{{.value.serviceName}} {{.value.servicePort}}:{{.value.servicePort}} > /dev/null 2>&1 &  curl localhost:{{.value.servicePort}}"
```
No need for the `script` here. It should be one only in the main test body (see the comment below).

aabouzaid commented on charts/camunda-platform-alpha/test/integration/testsuites/vars/files/testsuite-core.yaml:
```diff
@@ -305,3 +311,62 @@ testcases:
+      serviceName: camunda-keycloak
+      servicePort: 18080
+      script: "kubectl port-forward svc/{{.value.serviceName}} {{.value.servicePort}}:{{.value.servicePort}} > /dev/null 2>&1 &  curl localhost:{{.value.servicePort}} "
+    script: "{{.value.script}} "
```
Just make this line like this:

```
script: |
  kubectl port-forward svc/{{.value.serviceName}} {{.value.servicePort}}:{{.value.servicePort}} > /dev/null 2>&1 & \
  curl localhost:{{.value.servicePort}}
```

aabouzaid commented on charts/camunda-platform-alpha/test/integration/testsuites/vars/files/testsuite-core.yaml:
```diff
@@ -13,6 +13,12 @@ vars:
 testcases:
 
 - name: Core test suite is disabled
+  # https://docs.camunda.io/docs/self-managed/identity/user-guide/generating-m2m-tokens/
```
No need for those new lines.

aabouzaid commented on charts/camunda-platform-alpha/test/integration/testsuites/vars/files/testsuite-core.yaml:
```diff
@@ -305,3 +311,62 @@ testcases:
+  - name: "Port forward to {{ .value.component }} service "
+    type: exec
+    range:
+    - component: zeebe-gateway
```
Write the component name like this `Zeebe Gateway` (the same for all components, capitalized).

aabouzaid reviewed (changes requested):
We are now in the right direction. :clap: 
I've left a couple of notes.

aabouzaid commented on charts/camunda-platform-alpha/test/integration/testsuites/vars/files/testsuite-core.yaml:
```diff
@@ -305,3 +311,62 @@ testcases:
+#    retry: 2
+#    delay: 1
+  ########################################################Port forwarding Camunda components##################################
+- name: Port forwarding Camunda components
```
When we are done, please copy all changes to other versions:

```
charts/camunda-platform-8.3/test/integration/testsuites/vars/files/testsuite-core.yaml
charts/camunda-platform-8.4/test/integration/testsuites/vars/files/testsuite-core.yaml
charts/camunda-platform-8.5/test/integration/testsuites/vars/files/testsuite-core.yaml
charts/camunda-platform-8.6/test/integration/testsuites/vars/files/testsuite-core.yaml
```

mabdelgowa commented:
@aabouzaid 
check it right now
All components were capitalized, unnecessary lines were deleted, and only one line was for the script. Also, all changes were added to all versions.

aabouzaid commented:
@mabdelgowa Thanks for your effort in this PR. I'm closing it as we will use another test suite, and any changes in the current suite will not be relevant.
